### PR TITLE
add prospectus global 404 no-cache

### PR DESCRIPTION
**Pull request by fredsmith**

### CODE REVIEW

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2 [hunk 1]
```diff
@@ -53,7 +61,6 @@ server {
       auth_basic_user_file  {{ nginx_htpasswd_file }};
     {% endif %}
     port_in_redirect off;
-    error_page 404 /404.html;
```
Comment (rgraber) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2:56]: this we need for gatsby!reasons

**Review by rgraber — DISMISSED** [review]
I think we need the `error_page 404 /404.html` directives because of how gatsby does that page. the cache control thing i don't know about

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2 [hunk 2]
```diff
@@ -105,7 +109,6 @@ server {
 
     add_header 'Cache-Control' 'public, max-age=0, must-revalidate';
     try_files $uri $uri/index.html =404;
-    error_page 404 /404.html;
   }
 
   # PROSPECTUS_STATIC_SITES will be a list of dictionaries which have a:
```
Comment (tuchfarber) [inline playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2:114]: If we're in the spirit of ripping things out, we should probably rank this copy/pasted test code.

Comment (fredsmith) [reply playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2:114]: I just merged that last week!

Comment (fredsmith) [reply playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2:114]: Sorry, I mean, I just merged code that configures that last week.  https://github.com/edx/edx-internal/pull/3053

**Review by adzuci — APPROVED** [review]

File: playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2 (hunk 2 above)
Comment (tuchfarber) [reply playbooks/roles/nginx/templates/edx/app/nginx/sites-available/prospectus.j2:114]: But like... that regex was specific to the 2-layer learner portal MFE. I thought it was a test from a while ago that we abandoned. I have feelings about jamming this into prospectus, but it doesn't super matter as a test.

**Review by albemarle — APPROVED** [review]

**Pull request merged**